pyscriptjs/tests/support.py

```diff
@@ -0,0 +1,319 @@
+        substring), use self.page.expect_console_message directly.
+
+        timeout is expressed in milliseconds. If it's None, it will use
+        playwright's own default value, which is 30 seconds).
+
+        If check_errors is True (the default), it also checks that no JS
+        errors were raised during the waiting.
+        """
+        pred = lambda msg: msg.text == text
+        try:
+            with self.page.expect_console_message(pred, timeout=timeout):
+                pass
+        finally:
+            # raise JsError if there were any javascript exception. Note that
+            # this might happen also in case of a TimeoutError. In that case,
+            # the JsError will shadow the TimeoutError but this is correct,
+            # because it's very likely that the console message never appeared
+            # precisely because of the exception in JS.
+            if check_errors:
+                self.check_errors()
+
+    def wait_for_pyscript(self, *, timeout=None, check_errors=True):
+        """
+        Wait until pyscript has been fully loaded.
+
+        Timeout is expressed in milliseconds. If it's None, it will use
+        playwright's own default value, which is 30 seconds).
+
+        If check_errors is True (the default), it also checks that no JS
+        errors were raised during the waiting.
+        """
+        # this is printed by pyconfig.ts:PyodideRuntime.initialize
+        self.wait_for_console(
+            "===PyScript page fully initialized===",
+            timeout=timeout,
+            check_errors=check_errors,
+        )
+
+    def pyscript_run(self, snippet):
+        """
+        Main entry point for pyscript tests.
+
+        snippet contains a fragment of HTML which will be put inside a full
+        HTML document. In particular, the <head> automatically contains the
+        correct <script> and <link> tags which are necessary to load pyscript
+        correctly.
+
+        This method does the following:
+          - write a full HTML file containing the snippet
+          - open a playwright page for it
+          - wait until pyscript has been fully loaded
+        """
+        doc = f"""
+        <html>
+          <head>
+              <link rel="stylesheet" href="{self.http_server}/build/pyscript.css" />
+              <script defer src="{self.http_server}/build/pyscript.js"></script>
+          </head>
+          <body>
+            {snippet}
+          </body>
+        </html>
+        """
+        filename = f"{self.testname}.html"
+        self.writefile(filename, doc)
+        self.goto(filename)
+        self.wait_for_pyscript()
+
+
+# ============== Helpers and utility functions ==============
+
+
+class JsError(Exception):
+    """
+    Represent an exception which happened in JS.
+
+    It's a thin wrapper around playwright.sync_api.Error, with two important
+    differences:
+
+    1. it has a better name: if you see JsError in a trabeback, it's
```

Comment:
not to nitpick but noticed a small spelling mistake. `trabeback` --> `traceback`

Reply:
thank you! Fixed by 178e73c